### DISCUSSION TITLE
How do you request a `readMemory` call?

### DISCUSSION THREAD

**Discussion by ColinKennedy** [mfussenegger/nvim-dap#1084] [General]
In an active nvim-dap session, I tried calling

```
lua print(require("dap").session():request("readMemory", {memoryReference="0x320", count=8}))
```

Where `"0x320"` is the address of the `rax` register and I was going to read 8 bytes sequentially from that point. I checked the client and `supportsReadMemoryRequest = true,` so it seemed like it should work. But no matter how I call it, it returns `nil`. I can't tell if this is a bug or if my command is wrong (probably my command). Any advice would be appreciated!

**Comment by mfussenegger** [comment]
The function needs to run within a coroutine to return a result. Otherwise you'll need to provide a third argument that's a callback function with `err, result` as parameters.

I've added some docs for the request function: https://github.com/mfussenegger/nvim-dap/pull/1085

**Reply by ColinKennedy** [reply]
That worked, thank you for the quick reply and for the added docs!
